workflows/createtaxdb.nf

```diff
@@ -70,20 +70,54 @@ workflow CREATETAXDB {
                 fasta_dna
             }
 
+        // Make channel to preserve meta for decompress/compression
+        ch_dna_refs_for_rematching = ch_samplesheet.map { meta, fasta_dna, _fasta_aa ->
+            [
+                fasta_dna.getBaseName(fasta_dna.name.endsWith('.gz') ? 1 : 0),
+                meta,
+            ]
+        }
+
+        // Separate files for zipping and unzipping
         ch_dna_for_unzipping = ch_dna_refs_for_singleref.branch { _meta, fasta ->
             zipped: fasta.extension == 'gz'
             unzipped: true
         }
 
-        GUNZIP_DNA(ch_dna_for_unzipping.zipped)
-        ch_prepped_dna_fastas_ungrouped = GUNZIP_DNA.out.gunzip.mix(ch_dna_for_unzipping.unzipped)
-        ch_prepped_dna_fastas = ch_prepped_dna_fastas_ungrouped.map { _meta, fasta -> [[id: params.dbname], fasta] }.groupTuple()
-        ch_versions = ch_versions.mix(GUNZIP_DNA.out.versions.first())
+        // Batch the zipped files for efficient unzipping of multiple files in a single process job
+        ch_dna_batches_for_unzipping = ch_dna_for_unzipping.zipped
+            .map { _meta, fasta -> fasta }
+            .collate(params.unzip_batch_size, true)
```

Comment:
Something that came up the other day in nf-core/taxprofiler that has just come to my mind: will this collate/batching have the same order every time? If not, it may break resume which would not be ideal.

@muniheart has made a PR to address a similar batching problem in taxprofiler https://github.com/nf-core/taxprofiler/pull/576 which I've not properly looked at yet, but maybe we should do this here too for safety?

Reply:
Are we 100% sure that `.collate()` is non-deterministic, it seems strange that it wouldn't just emit slices of the input channel?